server/services/SlackBotService.ts

```diff
@@ -0,0 +1,49 @@
+import { Service } from "typedi";
+import { WebClient } from "@slack/web-api";
+import { SlackConfig } from "../Config";
+import { RoomsService } from "./RoomsService";
+import { RoomEvent } from "../express/types/RoomEvent";
+import { logger } from "../log";
+
+@Service({ multiple: false })
+export class SlackBotService {
+  private slackClient;
+
+  private lastNotificationTime: { [key: string]: number } = {};
+
+  constructor(private readonly roomsService: RoomsService) {}
+
+  init(config: SlackConfig) {
+    if (config.botOAuthAccessToken && !this.slackClient) {
+      this.slackClient = new WebClient(config.botOAuthAccessToken);
+      this.roomsService.listenRoomChange((event) => this.onRoomEvent(event));
+    }
+  }
+
+  private onRoomEvent(event: RoomEvent) {
+    if (event.type === "join") {
```

Comment:
Do you really want to post a message on every join or just when the first user enters / the last user leaves?